beamer/state_machine.py

```diff
@@ -179,20 +196,30 @@ def _handle_claim_made(event: ClaimMade, context: Context) -> bool:
         claim = Claim(event, challenge_back_off_timestamp)
         context.claims.add(claim.id, claim)
 
-        return True
+        return True, events
 
     # this is at least the second ClaimMade event for this claim id
     assert event.challenger != ADDRESS_ZERO, "Second ClaimMade event must contain challenger"
     try:
         # Agent is not part of ongoing challenge
         if context.address not in {event.claimer, event.challenger}:
             claim.ignore(event)
+
         claim.challenge(event)
     except TransitionNotAllowed:
-        return False
+        return False, events
+
+    if request.l1_resolution_tx_handle is None and request.fill_tx is not None:
```

Comment:
If I understand this correctly, whenever a claim is challenged this event is emitted everytime a new claimMade event arrives.

`_handle_initiate_l1_resolution()` never returns True, so each event will stay in the event loop forever. See the other comment at `_handle_initiate_l1_resolution()`